core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -214,75 +214,66 @@ protected MetadataResource putResource(final ServiceContext context, final Strin
-        }
+            doc = (Document) cmisConfiguration.getClient().getObjectByPath(key, oc);
 
-        Map<String, Object> properties = new HashMap<String, Object>();
-        if (!StringUtils.isEmpty(cmisConfiguration.getCmisMetadataUUIDPropertyName()) && !cmisConfiguration.existMetadataUUIDSecondaryProperty()) {
-            setCmisMetadataUUIDPrimary(properties, metadataUuid);
+            // Update existing document
+            setCmisProperties(metadataUuid, properties, doc, additionalProperties);
+            doc = cmisUtils.saveDocument(key, doc, properties, is, changeDate);
+        } catch (CmisObjectNotFoundException e) {
+            // add new document
+            setCmisProperties(metadataUuid, properties, null, additionalProperties);
+            doc = cmisUtils.saveDocument(key, null, properties, is, changeDate);
         }
 
-        // If cmisObject is empty then it is a new record. With new records, we need to set the default value for the status.
-        MetadataResourceExternalManagementProperties.ValidationStatus defaultStatus = null;
-        // We only need to set the default if there is a status property supplied.
-        if (!StringUtils.isEmpty(cmisConfiguration.getExternalResourceManagementValidationStatusPropertyName())) {
-            if (cmisConfiguration.getExternalResourceManagementValidationStatusDefaultValue() != null) {
-                // If a default property name does exist then use it
-                defaultStatus = MetadataResourceExternalManagementProperties.ValidationStatus.valueOf(cmisConfiguration.getExternalResourceManagementValidationStatusDefaultValue());
-            } else {
-                // Otherwise lets default to incomplete.
-                // Reason - as the administrator decided to use the status, it most likely means that there are extra properties that need to be set after a file is uploaded so defaulting it to
-                // incomplete seems reasonable.
-                defaultStatus = MetadataResourceExternalManagementProperties.ValidationStatus.INCOMPLETE;
-            }
-            if (cmisObject==null &&
-                !cmisConfiguration.existExternalResourceManagementValidationStatusSecondaryProperty() &&
-                !properties.containsKey(cmisConfiguration.getExternalResourceManagementValidationStatusPropertyName())
-                ) {
-                setCmisExternalManagementResourceStatusPrimary(properties, defaultStatus);
+        return createResourceDescription(context, metadataUuid, visibility, filename,
+            doc, metadataId, approved);
+    }
+
+    protected void setCmisProperties(String metadataUuid, Map<String, Object> properties, Document doc, Map<String, Object> additionalProperties) {
+
+        // Add additional properties if exists.
+        if (additionalProperties != null) {
+            if (MapUtils.isNotEmpty(additionalProperties)) {
+                properties.putAll(additionalProperties);
             }
         }
```

Comment:
```suggestion
        if (MapUtils.isNotEmpty(additionalProperties)) {
            properties.putAll(additionalProperties);
        }
```

https://commons.apache.org/proper/commons-collections/apidocs/org/apache/commons/collections4/MapUtils.html#isNotEmpty-java.util.Map-, indicates Null-safe check, no need to check for `null`